test/proj6api.jl

```diff
@@ -0,0 +1,69 @@
+using Test
+import Proj4
+
+@testset "Error handling" begin
+    @test Proj4.proj_errno_string(0) == nothing
+    @test Proj4.proj_errno_string(1) == "Operation not permitted"
+    @test Proj4.proj_errno_string(2) == "No such file or directory"
+
+    # this throws an internal error that is handled by our log_func
+    @test_throws Proj4.PROJError Proj4.proj_create("+proj=bobbyjoe")
+    # ensure we have reset the error state
+    @test Proj4.proj_context_errno() == 0
+
+end
+
+@testset "Database" begin
+    crs = Proj4.proj_create_from_database("EPSG", "4326", Proj4.PJ_CATEGORY_CRS, false)
+    Proj4.proj_errno(crs)
+    Proj4.proj_get_id_code(crs, 0)
+    # The following is wrong, but unfortunately segfaults. How to fix other than adding
+    # a null check before the ccall in this and other functions? PROJ documentation
+    # clearly states: must not be NULL
+    # Proj4.proj_get_id_code(C_NULL, 0)
```

Comment:
we should file this as an issue (to be resolved outside of this PR [after it gets merged])